backend/src/main/java/com/google/sps/datastore/ParticipantDatastore.java

```diff
@@ -57,40 +58,69 @@ public void addParticipant(Participant participant) {
     datastore.put(participantEntity);
   }
 
-  /** Return Participant Entity from username */
+  /** Return Participant Entity from username, or null if entity is not found */
+  @Nullable
   private Entity getEntityFromUsername(String username) {
     Key participantKey = KeyFactory.createKey(KIND_PARTICIPANT, username);
     try {
       return datastore.get(participantKey);
     } catch (EntityNotFoundException e) {
-      // TODO: what error to send
       return null;
     }
   }
 
-  /** Update Participant with new matchId, null out availability to show currently matched */
-  public void updateNewMatch(String username, long matchId) {
+  /**
+   * Update Participant entity with username with new matchId
+   *
+   * @return true if valid username (entity found), false if not
+   */
+  public boolean updateMatchId(String username, long matchId) {
     Entity participantEntity = getEntityFromUsername(username);
+    if (participantEntity == null) {
+      return false;
```

Comment:
How does the application get into this state?

Reply:
If `getParticipantFromUsername()` is called, since it returns `getParticipantFromEntity(getEntityFromUsername(username)`, and getEntityFromUsername(username) returns null if an entity is not found for that username.